solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/CollectionPropertiesZkStateReader.java

```diff
@@ -77,12 +75,13 @@ public class CollectionPropertiesZkStateReader implements Closeable {
   private final ExecutorService notifications =
       ExecutorUtil.newMDCAwareCachedThreadPool("cachecleaner");
 
-  // only kept to identify if the cleaner has already been started.
-  private Future<?> collectionPropsCacheCleaner;
+  // identify if the cleaner has already been started.
+  private final AtomicBoolean collectionPropsCacheCleanerInitialized = new AtomicBoolean(false);
```

Comment:
Yes - I was wondering using an ExecutorService is easier,  because of the number of lines of code added to stop the standalone thread when CPZSR closes. Let me know.

Reply:
it'd just be an interrupt & join (2 lines); no?

Reply:
Yes, done :). Visible in the latest commit.